src/main/java/seedu/cakecollate/logic/parser/FindCommandParser.java

```diff
@@ -1,33 +1,89 @@
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_ORDER_DESCRIPTION;
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_REQUEST;
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_TAG;
 
 import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import seedu.cakecollate.logic.commands.FindCommand;
 import seedu.cakecollate.logic.parser.exceptions.ParseException;
-import seedu.cakecollate.model.order.NameContainsKeywordsPredicate;
+import seedu.cakecollate.model.order.ContainsKeywordsPredicate;
 
 /**
  * Parses input arguments and creates a new FindCommand object
  */
 public class FindCommandParser implements Parser<FindCommand> {
 
+    private final Prefix[] allPrefixes = new Prefix[] {PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
+        PREFIX_ORDER_DESCRIPTION, PREFIX_TAG, PREFIX_DATE, PREFIX_REQUEST, PREFIX_DELIVERY_STATUS};
+
     /**
      * Parses the given {@code String} of arguments in the context of the FindCommand
      * and returns a FindCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(
+                args, allPrefixes);
+        List<Prefix> prefixes = Arrays.asList(allPrefixes);
+        HashMap<Prefix, List<String>> prefixesToFind = new HashMap<>();
+
+        // User input at least 1 prefix
+        if (!arePrefixesEmpty(argMultimap, allPrefixes)) {
+            String preamble = argMultimap.getPreamble();
+            if (!preamble.trim().isEmpty()) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            }
+
+            prefixes.forEach(prefix -> {
+                List<String> values = argMultimap.getAllValues(prefix);
+                if (!values.isEmpty()) {
+                    List<String> processed = processAllKeywords(values);
+                    prefixesToFind.put(prefix, processed);
+                }
+            });
+        // User did not input any prefix
+        } else {
+            String trimmedArgs = args.trim();
+            if (trimmedArgs.isEmpty()) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            }
+            List<String> processed = Arrays.asList(trimmedArgs.split("\\s+"));
+            prefixesToFind.put(new Prefix("all/"), processed);
```

Comment:
Might want to explain the purpose of this new prefix as a comment, or maybe in the DG. Took me a bit to follow, but seems like a good design choice to have this prefix 👍